docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -33,29 +33,106 @@ Note: Any build with incremental compilation disabled invalidates incremental ca
 
-> The new approach to incremental compilation is [Experimental](components-stability.md). It may be dropped or changed at any time.
-> Opt-in is required (see the details below). We encourage you to use it only for evaluation purposes, and we would
-> appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
->
-{type="warning"}
-
 The new approach to incremental compilation is available since Kotlin 1.7.0 for the JVM backend in the Gradle build system only. 
-This approach supports changes made inside dependent non-Kotlin modules, includes an improved
-compilation avoidance, and is compatible with the [Gradle build cache](#gradle-build-cache-support).
+Starting from Kotlin 1.8.20, it works by default. This approach supports changes made inside dependent non-Kotlin modules, 
+includes an improved compilation avoidance, and is compatible with the [Gradle build cache](#gradle-build-cache-support).
 
 All of these enhancements decrease the number of non-incremental builds, making the overall compilation time faster. 
 You will receive the most benefit if you use the build cache, or, frequently make changes in non-Kotlin
 Gradle modules.
 
-To enable this new approach, set the following option in your `gradle.properties`:
+To opt-out from this new approach, set the following option in your `gradle.properties`:
 
 ```none
-kotlin.incremental.useClasspathSnapshot=true
+kotlin.incremental.useClasspathSnapshot=false
 ```
 
+We'd appreciate your feedback on this feature in [YouTrack](https://youtrack.jetbrains.com/issue/KT-49682).
+
 Learn how the new approach to incremental compilation is implemented under the hood in
 [this blog post](https://blog.jetbrains.com/kotlin/2022/07/a-new-approach-to-incremental-compilation-in-kotlin/).
 
+### Precise backup of compilation tasks' outputs
+
+> Precise backup of compilation tasks' outputs is [Experimental](components-stability.md#stability-levels-explained).
+> We would appreciate your feedback on it in [YouTrack](https://kotl.in/issue/experimental-ic-optimizations).
+>
+{type="warning"}
+
+Starting with Kotlin 1.8.20, you can enable precise backup, whereby only those classes that Kotlin recompiles in 
+the incremental compilation will be backed up. Both full and precise backups help to run builds incrementally again 
+after compilation errors. Precise backup also saves build time compared to full backup. 
+Full backup may take **noticeable** build time in large projects or if many tasks are making backups, 
+especially if a project is located on a slow HDD.
+
+This optimization is Experimental. You can enable it by adding the `kotlin.compiler.preciseCompilationResultsBackup` 
+Gradle property to the `gradle.properties` file:
+
+`kotlin.compiler.preciseCompilationResultsBackup=true`
+
+#### Example of precise backup usage at JetBrains {initial-collapse-state="collapsed"}
+
+In the following chart, you can see an example of precise backup usage compared to full backup:
```

Comment:
What does the y axis of these charts represent? Is it build time for KGP and then Space?

Reply:
> Is it build time for KGP and then Space?

Yes, it is. But it's not for the whole project, it's for the build's subgraph: a module with changes is rebuilt + all other modules that depend on this module.